src/VisualStudio.Roslyn.SDK/ComponentDebugger/CapabilityProvider.cs

```diff
@@ -14,34 +14,30 @@ namespace Roslyn.ComponentDebugger
     [AppliesTo(ProjectCapabilities.CSharp + " | " + ProjectCapabilities.VB)]
     public class CapabilityProvider : ConfiguredProjectCapabilitiesProviderBase
     {
+        private readonly IProjectSnapshotService snapshotService;
+
         [ImportingConstructor]
         [System.Obsolete("This exported object must be obtained through the MEF export provider.", error: true)]
-        public CapabilityProvider(ConfiguredProject configuredProject)
+        public CapabilityProvider(ConfiguredProject configuredProject, IProjectSnapshotService snapshotService)
             : base(nameof(CapabilityProvider), configuredProject)
         {
+            this.snapshotService = snapshotService;
         }
 
         protected override async Task<ImmutableHashSet<string>> GetCapabilitiesAsync(CancellationToken cancellationToken)
         {
             // an alternative design could be to have 'IsRoslynComponent' just define the <Capability... directly in the managed.core targets
             // but that would require a specific roslyn version to work, this allows it to be backwards compatible with older SDKs
             var caps = Empty.CapabilitiesSet;
-            if (await IsRoslynComponentAsync(this.ConfiguredProject, cancellationToken).ConfigureAwait(false))
+
+            var snapshot = await snapshotService.GetLatestVersionAsync(ConfiguredProject, cancellationToken: cancellationToken).ConfigureAwait(false);
```

Comment:
Thanks @davkean and @tmeschter. I switched the cap provider over to this, which is the same way the 'get command line args' stuff works. Is that a valid way to handle getting it via dataflow?

Reply:
No. This needs to chain similar to what I pointed out here: https://github.com/dotnet/project-system/blob/feature/AspNet/src/Microsoft.VisualStudio.ProjectSystem.Managed.VS/ProjectSystem/VS/Web/ProjectTypeGuidsDataSource.cs. 

These systems flow data from one data source -> another, you need to transform the data as it flows along the pipe. This code just grabs the latest version from the snapshot pipe and there's sets up no relationship between the pipes.

Reply:
But isn't that handled by the base class? It sets up the pipe stuff and calls this method as it changes? I guess thats the bit I don't understand.

Reply:
No. Here you basically saying "I am the start of a pipe, and I will tell you when there are changes". What you want to say is "My pipe is based on the snapshot data, so when it changes, call me".

Reply:
This stuff is complicated and hard to get right - I can sit down and write it with you.